### PR TITLE
desktop: Use ashpd for dbus communication

### PR DESCRIPTION
**Pull request by kjarosh**
That's a refactor that simplifies our code a lot.

I would appreciate some tests to make sure it does not break anything.

### CODE REVIEW

File: desktop/src/gui/theme.rs [hunk 1]
```diff
@@ -171,10 +160,13 @@ impl ThemeController {
 }
 
 #[cfg(target_os = "linux")]
-fn scheme_to_theme(color_scheme: ColorScheme) -> Theme {
-    use crate::dbus::ColorScheme;
+fn scheme_to_theme(color_scheme: ashpd::desktop::settings::ColorScheme) -> Theme {
+    use ashpd::desktop::settings::ColorScheme;
     match color_scheme {
-        ColorScheme::Default => Theme::Light,
+        // We do not select the theme of window decorations,
```
Comment (torokati44) [inline desktop/src/gui/theme.rs:166]: Could you please clarify what you mean by this?

Comment (torokati44) [reply desktop/src/gui/theme.rs:166]: I mean... we don't select it "here", we don't _want_ to select it, we "can't" select it...?

Comment (torokati44) [reply desktop/src/gui/theme.rs:166]: Note, on a Wayland compositor with no support for server-side decorations, I believe we have the `sctk` decorations, which does have a dark mode:
https://github.com/PolyMeilex/sctk-adwaita

Comment (kjarosh) [reply desktop/src/gui/theme.rs:166]: > Could you please clarify what you mean by this?
>
> I mean... we don't select it "here", we don't _want_ to select it, we "can't" select it...?

It means that we just don't do it. We *can* do it for sure, but I haven't checked whether it is feasible or how easy it is.

When we don't select it, it means we have to assume it's light in order not to desync decorations with GUI. There are apps that use dark mode without dark preference, and will use only light mode when strictly required (e.g. GNOME terminal, it's dark even when dark mode is disabled).

Note that this does not change the behavior, just adds a comment explaining the situation.

Comment (kjarosh) [reply desktop/src/gui/theme.rs:166]: Turns out it's fairly simple: https://github.com/ruffle-rs/ruffle/pull/18213

File: desktop/src/dbus.rs [hunk 2]
```diff
@@ -4,83 +4,28 @@
...
-    Default,
-    PreferLight,
-    PreferDark,
-}
-
-#[derive(thiserror::Error, Debug)]
-pub enum FreedesktopSettingsError {
-    #[error("Unexpected value for color scheme: {0}")]
-    UnexpectedColorScheme(Value<'static>),
-}
```
Comment (torokati44) [inline desktop/src/dbus.rs:37]: Yay for less boilerplate!

**Review by torokati44 — APPROVED** [review]
The code looks fine, I appreciate the simplification.
I assume you have tested it and nothing regressed.
Seems like the `ashpd` release cadence is a bit "meh" _(no worries, it's FOSS after all!)_, and at least this lessens the likelihood of `zbus` dep duplication due to pulling it in both directly and indirectly.
Thank you! ^^

**Comment by torokati44** [comment]
This is on this branch - beautiful!

https://github.com/user-attachments/assets/226c20b2-0e00-49c8-9b22-946ec7a9faef

**Pull request merged**